findme-users/src/UserRepository.py

```diff
@@ -26,22 +27,48 @@ def update_user_in_db(self, user):
         return self.__put_user_to_db(user)
 
     def get_user_by_user_id_from_db(self, user_id):
-        if not self.__does_user_with_user_id_exist(user_id):
+        response = self.table.query(
+            IndexName="UserIdIndex",
+            ProjectionExpression="user_id, username, first_name, last_name",
+            KeyConditionExpression=Key('user_id').eq(user_id)
+        )
+        if not response.get('Items') or not response.get('Items')[0]:
             raise NotFoundError(f"No User with user_id: {user_id} found")
+        try:
+            return User(**response["Items"][0])
+        except ValidationError as e:
+            print(e)
+            raise BadRequestError(f"Unable to read Data from DB {e}")
+
+    def get_users_by_username_prefix(self, username_prefix):
+        # Handle the case where no username prefix is provided.
+        if not username_prefix:
+            return []
 
-        response = self.table.get_item(Key={'user_id': user_id})
         try:
-            return User(**response["Item"])
+            response = self.table.query(
+                ProjectionExpression="user_id, username, first_name, last_name",
+                KeyConditionExpression=Key('partition_key').eq("USER") & Key('username').begins_with(username_prefix)
+            )
+
         except ValidationError as e:
             print(e)
             raise BadRequestError(f"Unable to read Data from DB {e}")
 
+        return response.get('Items', [])
+
     def __does_user_with_user_id_exist(self, user_id):
-        return 'Item' in self.table.get_item(Key={'user_id': user_id})
+        return 'Item' in self.table.query(
```

Comment:
I think the response of this query will look something like this:
```json
{
    "Items": [
        {
            "last_name": {
                "S": "test2"
            },
            "user_id": {
                "S": "660d4d52722c22dc8212e5da"
            },
            "first_name": {
                "S": "test"
            },
            "username": {
                "S": "flyingbeat"
            }
        }
    ],
    "Count": 1,
    "ScannedCount": 1,
    "ConsumedCapacity": null
}
```
So probably 'Item' in ... will always return false?

Reply:
Fixed in ff0256c2b9365ebc12cd532583fb6dd8a270c10d, please confirm